Gems/AtomLyIntegration/CommonFeatures/Code/Source/PostProcess/EditorModeFeedback/EditorEditorModeFeedbackSystemComponent.cpp

```diff
@@ -0,0 +1,268 @@
+            AzToolsFramework::ViewportEditorMode mode)
+        {
+            // Purge the draw packets for all registered 
+            if (mode == AzToolsFramework::ViewportEditorMode::Focus)
+            {
+                for (auto& [entityId, componentMeshHandleDrawPackets] : m_entityComponentMeshHandleDrawPackets)
+                {
+                    for (auto& [componentid, meshHandleDrawPackets] : componentMeshHandleDrawPackets)
+                    {
+                        meshHandleDrawPackets.m_meshDrawPackets.clear();
+                    }
+                }
+
+                m_enabled = true;
+            }
+        }
+
+        void EditorEditorModeFeedbackSystemComponent::OnEditorModeDeactivated(
+            [[maybe_unused]] const AzToolsFramework::ViewportEditorModesInterface& editorModeState,
+            AzToolsFramework::ViewportEditorMode mode)
+        {
+            if (mode == AzToolsFramework::ViewportEditorMode::Focus)
+            {
+                m_enabled = false;
+            }
+        }
+
+        void EditorEditorModeFeedbackSystemComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+        {
+            if (!m_enabled)
+            {
+                return;
+            }
+
+            const auto focusModeInterface = AZ::Interface<AzToolsFramework::FocusModeInterface>::Get();
+            if (!focusModeInterface)
+            {
+                return;
+            }
+
+            // TODO: see if there is a more reliable method of creating the required resources once the depender systems are initialized
+            if (!m_maskMaterial)
+            {
+                m_maskMaterial = CreateMaskMaterial();
+            }
+
+            // Build the draw packets (where required) for each registered component and add them to the draw list
+            const auto focusedEntityIds = focusModeInterface->GetFocusedEntities(AzToolsFramework::GetEntityContextId());
+            for (const auto& focusedEntityId : focusedEntityIds)
+            {
+                const auto entityComponentMeshHandleDrawPacketsIterator = m_entityComponentMeshHandleDrawPackets.find(focusedEntityId);
+                if (entityComponentMeshHandleDrawPacketsIterator == m_entityComponentMeshHandleDrawPackets.end())
+                {
+                    // No drawable data for this entity
+                    continue;
+                }
+
+                auto& [entityId, componentMeshHandleDrawPackets] = *entityComponentMeshHandleDrawPacketsIterator;
+
+                for (auto& [componentId, meshHandleDrawPackets] : componentMeshHandleDrawPackets)
+                {
+                    const auto scene = RPI::Scene::GetSceneForEntityId(entityId);
+                    if (meshHandleDrawPackets.m_meshDrawPackets.empty())
+                    {
+                        if (const auto featureProcessor =
```

Comment:
Since you're getting the scene already with GetSceneForEntityId, just use scene->GetFeatureProcessor<MeshFeatureProcessorInterface>().

There is 1 feature processor of a given type per-scene. GetFeatureProcessorForEntity just calls GetSceneForEntityId under the hood, so it's redundant at this point.

Reply:
Fixed 👍